lib/bap/bap_self.ml

```diff
@@ -184,26 +197,40 @@ module Create() = struct
             Stream.unsubscribe Plugins.events subscription
           | _ -> () )
 
-
-    let bool = Arg.bool
-    let char = Arg.char
-    let int = Arg.int
-    let nativeint = Arg.nativeint
-    let int32 = Arg.int32
-    let int64 = Arg.int64
-    let float = Arg.float
-    let string = Arg.string
-    let enum = Arg.enum
-    let file = Arg.file
-    let dir = Arg.dir
-    let non_dir_file = Arg.non_dir_file
-    let list = Arg.list
-    let array = Arg.array
-    let pair = Arg.pair
-    let t2 = Arg.t2
-    let t3 = Arg.t3
-    let t4 = Arg.t4
-    let some = Arg.some
+    let converter : 'a Arg.converter -> 'a -> 'a converter =
+      fun (parser, printer) default -> parser, printer, default
+
+    let bool = converter Arg.bool false
+    let char = converter Arg.char '\x00'
+    let int = converter Arg.int 0
+    let nativeint = converter Arg.nativeint Nativeint.zero
+    let int32 = converter Arg.int32 Int32.zero
+    let int64 = converter Arg.int64 Int64.zero
+    let float = converter Arg.float 0.
+    let string = converter Arg.string ""
```

Comment:
Well, the whole point of the defaults is to be able to use it in a simple way, and in my opinion, the most sane default for a string is an empty string `""`.